scripts/tft/models.py

```diff
@@ -0,0 +1,50 @@
+from enum import Enum
+from typing import Dict, List
+
+from envparse import env
+from pydantic import BaseModel, SecretStr
+
+
+class Fmf(BaseModel):
+    url: str
+    ref: str
+    name: str
+
+
+class Test(BaseModel):
+    fmf: Fmf
+
+
+class CPUArch(Enum):
+    x86_64: str = "x86_64"
+
+
+class Os(BaseModel):
+    compose: str
+
+
+class ArtifactType(Enum):
+    fedora_copr_build: str = "fedora-copr-build"
+
+
+class Artifact(BaseModel):
+    id: str
+    type: ArtifactType
+
+
+class Environment(BaseModel):
+    arch: CPUArch
+    os: Os
+    variables: Dict[str, SecretStr]
+    artifacts: List[Artifact]
+
+
+class TFTRequest(BaseModel):
+    api_key: SecretStr = SecretStr(env.str("TFT_API_KEY"))
```

Comment:
The TFT API key is leaking to the TFT request and anyone can see it, e.g. https://api.dev.testing-farm.io/v0.1/requests/1abf7ac9-f1e0-490f-b9b7-1cda0e61b2c4. We need to fix that. The one in the example is already removed as it's been leaked.

Reply:
we can't do better on our side. there is a relevant ticket on tft https://gitlab.com/testing-farm/general/-/issues/27

Reply:
@thrix ^^